scalatags/src-3/scalatags/stylesheet/StyleSheet.scala

```diff
@@ -0,0 +1,130 @@
+    Seq("." + c.name)
+  )
+}
+
+/** Inherit from me to define a stylesheet which you can use to define
+  * styles which get serialized to a `String`. Does not allow the use
+  * of cascading tag/class selectors; use [[CascadingStyleSheet]] for that.
+  */
+abstract class StyleSheet(implicit sourceName: sourcecode.FullName) {
+
+  /** The name of this CSS stylesheet. Defaults to the name of the trait,
+    * but you can override
+    */
+  def customSheetName: Option[String] = None
+
+  /** Converts the name of the [[StyleSheet]]'s, the name of the member, and
+    * any applicable pseudo-selectors into the name of the CSS class.
+    */
+  protected[this] def nameFor(memberName: String, pseudoSelectors: String) = {
+    customSheetName.getOrElse(
+      defaultSheetName.replace(".", "-")
+    ) + "-" + memberName + pseudoSelectors
+  }
+
+  /** Namespace that holds all the css pseudo-selectors, to avoid collisions
+    * with tags and style-names and other things.
+    */
+  protected[this] val & = new Selector
+
+  /** `*` in a CSS selector.
+    */
+  protected[this] object * extends Selector(Seq("*"))
+
+  /** Used to define a new, uniquely-named class with a set of
+    * styles associated with it.
+    */
+  protected[this] object cls extends Creator(Nil)
+  protected[this] class Creator(selectors: Seq[String])
+      extends PseudoSelectors[Creator] {
+    def pseudoExtend(s: String): Creator = new Creator(selectors :+ s)
+
+    /** Collapse the tree of [[StyleSheetFrag]]s into a single [[Cls]],
+      * side-effect all the output into the styleSheetText, and return the
+      * [[Cls]]
+      */
+    def apply(args: StyleSheetFrag*)(implicit name: sourcecode.Name): Cls = {
+      Cls(nameFor(name.value, ""), selectors, args)
+    }
+  }
+
+  /** The default name of a stylesheet, filled in with the [[StyleSheet]] implicit macro
+    */
+  protected[this] def defaultSheetName =
+    sourceName.value.replace(' ', '.').replace('#', '.')
+
+  /** All classes defined in this stylesheet, filled in with the [[StyleSheet]] implicit macro
+    */
+  protected[this] def initStyleSheet()(implicit
+      sourceClasses: SourceClasses[this.type]
+  ) =
+    allClasses0 = Some(() => sourceClasses.value(this))
+
+  private[this] var allClasses0: Option[() => Seq[Cls]] = None
+  def allClasses = allClasses0 match {
+    case Some(f) => f()
+    case None =>
+      throw new Exception(
+        "No CSS classes found on stylesheet " + this +
+          ". Did you forget to call `initStyleSheet()` in the body of the style sheet?"
+      )
+  }
+
+  def styleSheetText = allClasses.map(_.structure.stringify(Nil)).mkString("\n")
+}
+class SourceClasses[T <: StyleSheet](val value: T => Seq[Cls])
+object SourceClasses {
+  inline def apply[T <: StyleSheet]: SourceClasses[T] = $ { manglerImpl[T] }
+
+  def manglerImpl[T <: StyleSheet: Type](using Quotes): Expr[SourceClasses[T]] = {
+    ' { new SourceClasses[T]((t: T) => ${ Expr.ofSeq(terms('t)) }) }
```

Comment:
Github's code highlighting doesn't seem to be aware of the new quotes syntax 😉

Reply:
that would be a good bug report over at https://github.com/scala/vscode-scala-syntax/issues (repo name says "vscode", but it also powers highlighting on GitHub)

Reply:
Thank you for the suggestion @SethTisue 🙂 
I opened an issue as you suggested and opened also a PR with a fix.
Issue: https://github.com/scala/vscode-scala-syntax/issues/226, PR: https://github.com/scala/vscode-scala-syntax/pull/227

Reply:
oh wow even better!